crates/rethnet_evm_napi/src/state.rs

```diff
@@ -0,0 +1,303 @@
+        self.db
+            .checkpoint()
+            .await
+            .map_err(|e| napi::Error::new(Status::GenericFailure, e.to_string()))
+    }
+
+    #[napi]
+    pub async fn revert(&self) -> napi::Result<()> {
+        self.db
+            .revert()
+            .await
+            .map_err(|e| napi::Error::new(Status::GenericFailure, e.to_string()))
+    }
+
+    #[napi]
+    pub async fn get_account_by_address(&self, address: Buffer) -> napi::Result<Option<Account>> {
+        let address = Address::from_slice(&address);
+
+        self.db.account_by_address(address).await.map_or_else(
+            |e| Err(napi::Error::new(Status::GenericFailure, e.to_string())),
+            |account_info| Ok(account_info.map(Account::from)),
+        )
+    }
+
+    #[napi]
+    pub async fn get_account_storage_slot(
+        &self,
+        address: Buffer,
+        index: BigInt,
+    ) -> napi::Result<BigInt> {
+        let address = Address::from_slice(&address);
+        let index = BigInt::try_cast(index)?;
+
+        self.db
+            .account_storage_slot(address, index)
+            .await
+            .map_or_else(
+                |e| Err(napi::Error::new(Status::GenericFailure, e.to_string())),
+                |value| {
+                    Ok(BigInt {
+                        sign_bit: false,
+                        words: value.into_limbs().to_vec(),
+                    })
+                },
+            )
+    }
+
+    #[napi]
+    pub async fn get_code_by_hash(&self, code_hash: Buffer) -> napi::Result<Buffer> {
+        let code_hash = B256::from_slice(&code_hash);
+
+        self.db.code_by_hash(code_hash).await.map_or_else(
+            |e| Err(napi::Error::new(Status::GenericFailure, e.to_string())),
+            |code| Ok(Buffer::from(&code.bytes()[..code.len()])),
+        )
+    }
+
+    #[napi]
+    pub async fn get_state_root(&self) -> napi::Result<Buffer> {
+        self.db.state_root().await.map_or_else(
+            |e| Err(napi::Error::new(Status::GenericFailure, e.to_string())),
+            |root| Ok(Buffer::from(root.as_ref())),
+        )
+    }
+
+    #[napi]
+    pub async fn insert_account(&self, address: Buffer, account: Account) -> napi::Result<()> {
+        let address = Address::from_slice(&address);
+        let account = account.try_cast()?;
+
+        self.db
+            .insert_account(address, account)
+            .await
+            .map_err(|e| napi::Error::new(Status::GenericFailure, e.to_string()))
+    }
+
+    #[napi]
+    pub async fn make_snapshot(&self) -> Buffer {
+        <B256 as AsRef<[u8]>>::as_ref(&self.db.make_snapshot().await).into()
+    }
+
+    #[napi(ts_return_type = "Promise<void>")]
+    pub fn modify_account(
+        &self,
+        env: Env,
+        address: Buffer,
+        #[napi(
+            ts_arg_type = "(balance: bigint, nonce: bigint, code: Buffer | undefined) => Promise<AccountData>"
+        )]
+        modify_account_fn: JsFunction,
```

Comment:
Just to be sure I understand: this function is called with the current state of the account, and it updates the account with the values returned by the function. Is that correct? If so, can we document it? I believe just adding a `///` comment generates the proper jsdoc, like it's done in the `finalize` and `abort` functions of the block builder.

Reply:
Yes, your understanding is correct.

I added the documentation. Once the API solidifies I'll ensure that every function, object, and class has comments on it.